File: codegen/results/v002/m3-max-48gb/ollama/gemma4_31b-mlx_think/20260910_102409/typescript/008_prime_numbers/solution.ts
/**
 * This program accepts a single integer N from the command line 
 * and prints all prime numbers up to and including N, one per line.
 */

function main() {
    // Extract the argument from the command line
    const arg = process.argv[2];

    // If no argument is provided, we exit silently
    if (arg === undefined) {
        return;
    }

    // Parse the argument as an integer
    const N = parseInt(arg, 10);

    // Validate that the input is a number and at least 2 (the smallest prime)
    if (isNaN(N) || N < 2) {
        return;
    }

    /**
     * We use the Sieve of Eratosthenes algorithm to find all primes up to N.
     * This is highly efficient with a time complexity of O(N log log N).
     */
    
    // Create a boolean array (using Uint8Array for memory efficiency)
    // initialized to 1 (true), where index represents the number.
    const isPrime = new Uint8Array(N + 1).fill(1);
    
    // 0 and 1 are not prime numbers
    isPrime[0] = 0;
    isPrime[1] = 0;

    // Iterate from 2 up to the square root of N
    for (let p = 2; p * p <= N; p++) {
        // If isPrime[p] is still 1, then it is a prime
        if (isPrime[p] === 1) {
            // Mark all multiples of p starting from p*p as not prime
            for (let i = p * p; i <= N; i += p) {
                isPrime[i] = 0;
            }
        }
    }

    // Print all indices that remained marked as prime
    for (let p = 2; p <= N; p++) {
        if (isPrime[p] === 1) {
            process.stdout.write(p.toString() + '\n');
        }
    }
}

// Execute the main function
main();

File: codegen/results/v002/m3-max-48gb/ollama/gemma4_31b-mlx_think/20260910_102409/python/008_prime_numbers/solution.py
import sys

def print_primes(n):
    """
    Prints all prime numbers up to and including n using the Sieve of Eratosthenes.
    """
    if n < 2:
        return

    # Initialize a boolean array "is_prime[0..n]" 
    # and set all entries to True.
    is_prime = [True] * (n + 1)
    is_prime[0] = is_prime[1] = False  # 0 and 1 are not prime numbers

    p = 2
    while (p * p <= n):
        # If is_prime[p] is not changed, then it is a prime
        if is_prime[p]:
            # Update all multiples of p starting from p*p
            for i in range(p * p, n + 1, p):
                is_prime[i] = False
        p += 1

    # Print all prime numbers
    for p in range(2, n + 1):
        if is_prime[p]:
            print(p)

def main():
    # Check if the argument is provided
    if len(sys.argv) != 2:
        return

    try:
        # Convert the command line argument to an integer
        n = int(sys.argv[1])
        print_primes(n)
    except ValueError:
        # If the argument is not an integer, the program simply exits
        pass

if __name__ == "__main__":
    main()